shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1730,38 +1814,33 @@ - (void)beginFloatingCursorAtPoint:(CGPoint)point {
   //     caretRectForPosition boundingBox = self.convertRect(bounds, fromView:textInputView)
   //     bounds = self._selectionClipRect ?? self.bounds
   //
-  // It's tricky to provide accurate "bounds" and "caretRectForPosition" so it's preferred to
-  // bypass the clamping and implement the same clamping logic in the framework where we have easy
-  // access to the bounding box of the input field and the caret location.
-  //
-  // The current implementation returns kSpacePanBounds for "bounds" when
-  // "_isFloatingCursorActive" is true. kSpacePanBounds centers "caretRectForPosition" so the
-  // floating cursor has enough clearance in all directions to move around.
-  //
   // It seems impossible to use a negative "width" or "height", as the "convertRect"
   // call always turns a CGRect's negative dimensions into non-negative values, e.g.,
   // (1, 2, -3, -4) would become (-2, -2, 3, 4).
-  _isFloatingCursorActive = true;
+  _isFloatingCursorActive = YES;
+  _floatingCursorOffset = point;
   [self.textInputDelegate flutterTextInputView:self
                           updateFloatingCursor:FlutterFloatingCursorDragStateStart
                                     withClient:_textInputClient
-                                  withPosition:@{@"X" : @(point.x), @"Y" : @(point.y)}];
+                                  withPosition:@{@"X" : @0, @"Y" : @0}];
```

Comment:
maybe rename to `withRelativePosition`?

Reply:
why do we need this change?

Reply:
Before, `point` was always zero, because we always returned `CGRectZero` for caretRectForPosition. Now that we will get the actual point here, we need to make it relative as framework is written to expect points relative to zero.

Reply:
Can you point me to the corresponding framework side PR?

Reply:
There is no framework side PR, this change is to keep the same data going to the framework even as we change `caretRectForPosition`. For reference, see `EditableTextState.updateFloatingCursor`.